code/modules/mob/dead/observer/observer.dm

```diff
@@ -359,6 +359,11 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	abstract_move(destination) // move like the wind
 	return TRUE
 
+/mob/dead/observer/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	var/area/new_area = get_area(src)
+	update_ambience_area(new_area)
```

Comment:
```suggestion
	if(new_area != ambience_tracked_area)
		update_ambience_area(new_area)
```
This should save some useless proc calls.

Reply:
Good point, especially with something like Move()